vsphere/panels.libsonnet

```diff
@@ -0,0 +1,1118 @@
+          fieldOverride.byName.new('Memory ballooned')
+          + fieldOverride.byName.withPropertiesFromOptions(
+            table.standardOptions.withUnit('mbytes')
+          ),
+        ])
+        + table.standardOptions.withOverridesMixin([
+          fieldOverride.byName.new('Memory swapped')
+          + fieldOverride.byName.withPropertiesFromOptions(
+            table.standardOptions.withUnit('mbytes')
+          ),
+        ])
+        + table.standardOptions.withOverridesMixin([
+          fieldOverride.byName.new('Disk throughput')
+          + fieldOverride.byName.withPropertiesFromOptions(
+            table.standardOptions.withUnit('KiBs')
+          ),
+        ])
+        +
+        table.queryOptions.withTransformationsMixin([
+          {
+            id: 'joinByField',
+            options: {
+              byField: 'vcenter_host_name',
+              mode: 'outer',
+            },
+          },
+          {
+            id: 'filterFieldsByName',
+            options: {
+              include: {
+                names: [
+                  'vcenter_host_name',
+                  'Value #A',
+                  'Value #B',
+                  'Value #C',
+                  'Value #D',
+                  'Value #E',
+                  'vcenter_vm_name 1',
+                ],
+              },
+            },
+          },
+          {
+            id: 'organize',
+            options: {
+              excludeByName: {},
+              includeByName: {},
+              indexByName: {
+                'Value #A': 2,
+                'Value #B': 3,
+                'Value #C': 4,
+                'Value #D': 5,
+                'Value #E': 6,
+                vcenter_host_name: 0,
+                'vcenter_vm_name 1': 1,
+              },
+              renameByName: {
+                'Value #A': 'CPU utilization',
+                'Value #B': 'Memory utilization',
+                'Value #C': 'Memory ballooned',
+                'Value #D': 'Memory swapped',
+                'Value #E': 'Disk throughput',
+                'Value #F': 'Memory swapped',
+                vcenter_cluster_name: 'Cluster',
+                'vcenter_cluster_name 1': 'Cluster',
+                vcenter_host_name: 'Host',
+                'vcenter_host_name 1': 'Host',
+                vcenter_vm_name: 'Name',
+                'vcenter_vm_name 1': 'Name',
+                'vcenter_vm_name 2': '',
+              },
+            },
+          },
+        ]),
+
+      clusterCPUEffective:
+        barGauge.new(title='Cluster CPU effective')
+        + barGauge.queryOptions.withTargets([
+          t.clusterCPUEffective,
+        ])
+        + barGauge.panelOptions.withDescription('The effective CPU capacity of the cluster.')
+        + barGauge.options.withOrientation('horizontal')
+        + barGauge.standardOptions.withUnit('rotmhz')
+        + barGauge.standardOptions.color.withMode('continuous-YlRd'),
+
+      clusterCPULimit:
```

Comment:
The look of the two pairs of effective/limit panels is weird to me when the effective is all the way full, but the limit is empty, even though the limit >= effective.
<img width="981" alt="Screenshot 2024-05-07 at 3 01 43 PM" src="https://github.com/grafana/jsonnet-libs/assets/69878936/182fd331-8f85-4b97-bac7-5eaadd0ae153">

Reply:
I can see how it looks a bit weird, They are calculating the mix/max depending on what values looked like for that metric at different points in time. Since there is only one cluster it will skew the relative min/max calculations to base off values shown only for that cluster, which in this case shows the cluster cpu effective/memory effective to be "maxed" out on the gauge since there was previously a lower value on that metric for that singular cluster. I don't really see a way around this, its just how bar gauges work. If there were other clusters and more values then this will even itself out so to speak.
<img width="1331" alt="image" src="https://github.com/grafana/jsonnet-libs/assets/38274348/ef781b40-76bf-40fe-8879-7d8501d841f3">